packages/server/src/core/initTRPC.ts

```diff
@@ -44,9 +44,20 @@ type CreateRootConfigTypesFromPartial<TTypes extends PartialRootConfigTypes> =
  * - Doesn't need to be a class but it doesn't really hurt either
  */
 
+// If the type is a function (sync or async) unwrap it, else use the type as is
+type Unwrap<TType> = TType extends (...args: any[]) => infer R
+  ? Awaited<R>
```

Comment:
This requires TS 4.5 so we shouldn't do that one until next major

Reply:
I want to avoid us changing reqs of TS versions unless it enables big features

Reply:
used our custom `ThenArg` type for now